src/distrs/tdist.jl

```diff
@@ -13,7 +13,9 @@ import .RFunctions:
     tdistinvlogccdf
 
 # pdf for numbers with generic types
-tdistpdf(ν::Real, x::Number) = gamma((ν + 1) / 2) / (sqrt(ν * pi) * gamma(ν / 2)) * (1 + x^2 / ν)^(-(ν + 1) / 2)
+tdistpdf(ν::T, x::T) where T<:Real = gamma((ν + 1) / 2) / (sqrt(ν * π) * gamma(ν / 2)) * (1 + x^2 / ν)^(-(ν + 1) / 2)
+tdistpdf(ν::Real, x::Real) = tdistpdf(promote(ν, x)...)
 
 # logpdf for numbers with generic types
-tdistlogpdf(ν::Real, x::Number) = loggamma((ν + 1) / 2) - log(ν * pi) / 2 - loggamma(ν / 2) + (-(ν + 1) / 2) * log(1 + x^2 / ν)
+tdistlogpdf(ν::T, x::T) where T<:Real = loggamma((ν + 1) / 2) - log(ν * π) / 2 - loggamma(ν / 2) + (-(ν + 1) / 2) * log(1 + x^2 / ν)
```

Comment:
I guess here `log1p` could actually be useful:
```suggestion
tdistlogpdf(ν::T, x::T) where T<:Real = loggamma((ν + 1) / 2) - log(ν * π) / 2 - loggamma(ν / 2) + (-(ν + 1) / 2) * log1p(x^2 / ν)
```